operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependentResource.java

```diff
@@ -217,13 +212,7 @@ protected Resource<R> prepare(R desired, P primary, String actionName) {
         desired.getClass(),
         ResourceID.fromResource(desired));
 
-    addReferenceHandlingMetadata(desired, primary);
-
-    if (desired instanceof Namespaced) {
-      return client.resource(desired).inNamespace(desired.getMetadata().getNamespace());
```

Comment:
@csviri @metacosm this is only needed if a NamespacedKubernetesClient that has had inNamespace called is used as the client.  That shouldn't be the case correct?

Reply:
As long as that namespace is already populated on the resource, then that is what the client will use.  The namespacing cleanup in the client didn't happen until 6.0, so you may have had an issue before that.

Reply:
yep, ok great!